sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1188,6 +1188,12 @@ class Dataset[T] private[sql](
   def col(colName: String): Column = colName match {
     case "*" =>
       Column(ResolvedStar(queryExecution.analyzed.output))
+    case ParserUtils.escapedIdentifier(columnNameRegex)
```

Comment:
this is a breaking change, previously we always treat the input string as a column name(except for star), even it's quoted.